auth0/src/main/java/com/auth0/android/authentication/storage/SecureCredentialsManager.java

```diff
@@ -0,0 +1,251 @@
+    private boolean authenticateBeforeDecrypt;
+    private int authenticationRequestCode = -1;
+    private Activity activity;
+
+    //State for retrying operations
+    private BaseCallback<Credentials, CredentialsManagerException> decryptCallback;
+    private Intent authIntent;
+
+
+    @VisibleForTesting
+    SecureCredentialsManager(@NonNull AuthenticationAPIClient apiClient, @NonNull Storage storage, @NonNull CryptoUtil crypto) {
+        this.apiClient = apiClient;
+        this.storage = storage;
+        this.crypto = crypto;
+        this.gson = new Gson();
+        this.authenticateBeforeDecrypt = false;
+    }
+
+    /**
+     * Creates a new SecureCredentialsManager to handle Credentials
+     *
+     * @param context   a valid context
+     * @param apiClient the Auth0 Authentication API Client to handle token refreshment when needed.
+     * @param storage   the storage implementation to use
+     */
+    public SecureCredentialsManager(@NonNull Context context, @NonNull AuthenticationAPIClient apiClient, @NonNull Storage storage) {
+        this(apiClient, storage, new CryptoUtil(context, storage, KEY_ALIAS));
+    }
+
+    /**
+     * Require the user to authenticate using the configured LockScreen before accessing the credentials.
+     * This feature is disabled by default and will only work if the device is running on Android version 21 or up and if the user
+     * has configured a secure LockScreen (PIN, Pattern, Password or Fingerprint).
+     * <p>
+     * The activity passed as first argument here must override the {@link Activity#onActivityResult(int, int, Intent)} method and
+     * call {@link SecureCredentialsManager#checkAuthenticationResult(int, int)} with the received parameters.
+     *
+     * @param activity    a valid activity context. Will be used in the authentication request to launch a LockScreen intent.
+     * @param requestCode the request code to use in the authentication request. Must be a value between 1 and 200.
+     * @param title       the text to use as title in the authentication screen.
+     * @param description the text to use as description in the authentication screen. On some Android versions it might not be shown.
+     * @return whether this device supports requiring authentication or not.
+     */
+    public boolean requireAuthentication(@NonNull Activity activity, int requestCode, @Nullable String title, @Nullable String description) {
+        if (requestCode < 1 || requestCode > 255) {
+            throw new IllegalArgumentException("Request code must a value between 1 and 255.");
+        }
+        KeyguardManager kManager = (KeyguardManager) activity.getSystemService(Context.KEYGUARD_SERVICE);
+        this.authIntent = Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP ? kManager.createConfirmDeviceCredentialIntent(title, description) : null;
+        this.authenticateBeforeDecrypt = (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && kManager.isDeviceSecure()
```

Comment:
Readability would be better with parenthesis.  Think of parentheses like comments in code: they aren't strictly necessary, and in theory a competent developer should be able to figure out code without them. And yet, these cues are exceedingly helpful.

Reply:
I will add them but TBH it would almost look the same, as the new line is already doing that thing you mention 😛